### DISCUSSION TITLE
How can I fine-tune the table settings from the given pdf?

### DISCUSSION THREAD

**Discussion by donburi82** [jsvine/pdfplumber#867] [Ask for help with specific PDFs]
I am having some troubles when trying to fine-tune the table settings for the following pdf.

[document.pdf](https://github.com/jsvine/pdfplumber/files/11250453/document.pdf)

I first crop out the area containing the table better extraction using the code below:
cropped = p0.within_bbox((150, 123, p0.width-151, p0.height-190))
c = cropped.to_image()

![cropped](https://user-images.githubusercontent.com/104419950/232491830-b58b83a6-925a-44fb-a3f4-0ce324df054d.png)

Below is the best settings (for extracting the table) I have found so far but I am not sure how to further fine-tune the settings:

table_settings = {
    "vertical_strategy": "explicit",
    "horizontal_strategy": "explicit",
    "explicit_vertical_lines": p0.curves+p0.edges,
    "explicit_horizontal_lines": p0.curves+p0.edges,
    "intersection_tolerance": 15,
}

![best](https://user-images.githubusercontent.com/104419950/232491823-9bb0bc70-be29-472c-a667-f0e04d4cb7a7.png)

Any hints or help would be greatly appreciated.

**Comment by jsvine** [comment]
The issue here seems to be that the table does not include any vertical lines for the left and right borders. So you might try deriving those positions programmatically and passing them to `explicit_vertical_lines`. E.g.,:

```python
settings = {
  "explicit_vertical_lines": (
    [ r["x0"] for r in page.rects if r["width"] > 300 ] +
    [ r["x1"] for r in page.rects if r["width"] > 300 ]
  )
}

im = page.to_image(resolution=150)
im.debug_tablefinder(settings)
im.save(...)
```

Produces:

![tmp](https://user-images.githubusercontent.com/534702/233054294-099a0a17-7a85-4755-8635-1a573c43aa4a.png)

**Reply by donburi82** [reply]
This is great! Thank you so much for your help :D
